src/wallet/load.cpp

```diff
@@ -62,7 +62,7 @@ bool VerifyWallets(interfaces::Chain& chain)
     std::set<fs::path> wallet_paths;
 
     for (const auto& wallet_file : gArgs.GetArgs("-wallet")) {
-        const fs::path path = fs::absolute(wallet_file, GetWalletDir());
+        const fs::path path = fs::absolute(GetWalletDir() / wallet_file);
```

Comment:
I think this isn't equivalent if wallet_file is the empty string (this will add trailing slash now, wouldn't add trailing slash previously).

Reply:
Yes, this simple sample confirms it:

```cpp
// Run with: g++ --std=c++17 path.cpp && ./a.out

#include <string>
#include <iostream>
#include <filesystem>
namespace fs = std::filesystem;

int main()
{
    fs::path walletDir = "/home/user";
    std::string wallet_file = "";
    fs::path path = fs::absolute(walletDir / wallet_file);

    std::cout << "walletDir = " << walletDir << "\n"; // walletDir = "/home/user"
    std::cout << "path      = " << path << "\n";      // path      = "/home/user/"
}
```

Reply:
In commit "refactor: replace boost::filesystem with std::filesystem" (f449c26f30bb8e223d4d7833c579d555a0005daa)

re: https://github.com/bitcoin/bitcoin/pull/20744#discussion_r556553883

> Yes, this simple sample confirms it:

Issue should be resolved when this is rebased after #20932.

Reply:
This has been rebased post #20932, so marking this as resolved.